server/models/instance.py

```diff
@@ -70,6 +73,39 @@ def list(self, user=None, tale=None, limit=0, offset=0,
                 cursor=cursor, user=currentUser, level=AccessType.READ,
                 limit=limit, offset=offset):
             yield r
+            
+    def updateAndRestartInstance(self, instance, token, **kwargs):
```

Comment:
I'd rather avoid using `**kwargs` construct. Could you explicitly specify `imageId` and `digest` instead?

Reply:
Absolutely!